### PR TITLE
HDDS-6019. Fix s3g create key error in cluster with replication one.

### PR DESCRIPTION
**Pull request by guihecheng**
## What changes were proposed in this pull request?

Fix s3g create key error in cluster with replication one.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6019

## How was this patch tested?
manual test

### CODE REVIEW

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3StorageType.java [hunk 1]
```diff
@@ -48,8 +50,17 @@ public ReplicationType getType() {
     return type;
   }
 
-  public static S3StorageType getDefault() {
-    return STANDARD;
+  public static S3StorageType getDefault(ConfigurationSource config) {
```
Comment (bharatviswa504) [inline hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3StorageType.java:53]: Can we add JavaDoc for this.

And also what is the reason if config has wrong value, falling back to STANDARD?

**Review by bharatviswa504 — COMMENTED** [review]
Overall LGTM.
One minor comment/question.

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3StorageType.java (hunk 1 above)
Comment (guihecheng) [reply hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3StorageType.java:53]: @bharatviswa504 sure, I'll add more descriptions.
Yes, for now I just try to fallback to STANDARD.

**Review by bharatviswa504 — APPROVED** [review]

**Comment by bharatviswa504** [comment]
Thank You @guihecheng for the contribution.

**Pull request merged**